src/Microsoft.Azure.SignalR.Common/ServiceConnections/ConnectionFactory.cs

```diff
@@ -54,10 +54,14 @@ public async Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat,
             }
         }
 
-        private Uri GetServiceUrl(string connectionId)
+        private Uri GetServiceUrl(string connectionId, string target)
         {
             var baseUri = new UriBuilder(_provider.GetServerEndpoint(_hubName));
             var query = "cid=" + connectionId;
+            if (target != null)
+            {
+                query = query + $"&target={target}";
```

Comment:
url encode (data string)

Reply:
$"{query}&target={target}"

---
In reply to: [248543449](https://github.com/Azure/azure-signalr/pull/342#discussion_r248543449) [](ancestors = 248543449)